src/TraceEvent/EventPipe/EventMetadata.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.IO;
+
+namespace Microsoft.Diagnostics.Tracing.EventPipe
```

Comment:
I recommend that this type  be a IFastSerializable (since it is something in the file), and its FromStream read in its fields.   But I also don't recommend that it have most of the fields below.   Instead it should have the TraceEvent type for that meta-data set up that has been filled out (including a EVENT_TRACE_HEADER blob as well as the blob size needed for the payload.   Basically anything you can do before you get the event data should be done here.  

The idea is that to dispatch an events you only need to deserialize the event number, look up its EventMetatData, copy the right number of bytes from the ETLX file to to the buffer associated with the EventMetatData, and return the TraceEvent (super fast).